src/main/qss.js

```diff
@@ -415,20 +440,81 @@ gpii.app.qssWrapper.updateSettings = function (that, settings, notUndoable) {
     });
 };
 
+
+/**
+ * Generates a language setting option label out of a provided language metadata.
+ * @param {String} languageOptionLabelTemplate - The template for every language label
+ * @param {Object} languageMetadata - The language metadata that includes different labels
+ * for the language
+ * @param {Boolean} isMainLocale - We wouldn't want to display both the native representation
+ * and the main language as they are redundant
+ * @return {String} The desired language label
+ */
+gpii.app.qssWrapper.getLanguageLabel = function (languageOptionLabelTemplate, languageMetadata, isMainLocale) {
+    languageMetadata.local =
+        !languageMetadata.local ?
+            languageMetadata.english :
+            languageMetadata.local;
+
+    // Native labels might be coming in full lowercase
+    languageMetadata["native"] = gpii.flowManager.capitalizeFirstLetter(languageMetadata["native"]);
+
+    return isMainLocale ?
+        languageMetadata.local :
+        fluid.stringTemplate(languageOptionLabelTemplate, languageMetadata);
+};
+
+
+/**
+ * Generates a list of language labels out of the provided languages metadata.
+ * @param {Object} settingOptions - The QSS settings specific options
+ * @param {String} settingOptions.languageOptionLabelTemplate - The template for every language label
+ * @param {Object[]} installedLanguages - A list of all languages metadata
+ * @return {String[]} The list of labels
+ */
+gpii.app.qssWrapper.getLanguageLabels = function (settingOptions, installedLanguages) {
+    return installedLanguages.map(function (languageMetadata) {
+        return gpii.app.qssWrapper.getLanguageLabel(
+            settingOptions.languageOptionLabelTemplate,
+            languageMetadata,
+            languageMetadata.current
```

Comment:
This probably needs to be changed. In case we have multiple region options for a language, I suppose we wouldn't want to show the native representation of, lets say `English (Canada)` in case the current language is `English (United States)`